dotcom-rendering/playwright/tests/parallel-5/liveblog.interactivity.e2e.spec.ts

```diff
@@ -0,0 +1,306 @@
+	});
+
+	test('should enhance tweets after they have been inserted', async ({
+		page,
+	}) => {
+		await loadPage(page, `/Article/${blogUrl}?live=true`);
+		await waitForIsland(page, 'Liveness', { waitFor: 'attached' });
+
+		await page.evaluate((tweet) => {
+			window.mockLiveUpdate({
+				numNewBlocks: 1,
+				html: tweet,
+				mostRecentBlockId: 'abc',
+			});
+		}, tweetBlock);
+
+		const updateLocator = page
+			.getByText('1 new update')
+			.locator('visible=true');
+
+		await expect(updateLocator).toBeVisible();
+		await updateLocator.click();
+
+		await waitForIsland(page, 'TweetBlockComponent', {
+			waitFor: 'attached',
+		});
+
+		// have to use an attribute selector here as a numeric id is not a valid selector
+		const tweetContainerSelector =
+			'[id="46d194c9-ea50-4cd5-af8b-a51e8b15c65e"]';
+
+		await page
+			.locator(tweetContainerSelector)
+			.waitFor({ state: 'visible', timeout: 20000 });
+
+		await page.locator(tweetContainerSelector).scrollIntoViewIfNeeded();
+
+		const twitterIframe = getIframeBody(
+			page,
+			`${tweetContainerSelector} iframe`,
+		);
+
+		await expect(await twitterIframe).toContainText(
+			'They will prepare the extraordinary European Council meeting tonight',
+			{ timeout: 20000 },
+		);
+	});
+
+	test('should use the right block id when polling from the second page', async ({
+		page,
+	}) => {
+		const lastUpdateRequestPromise = page.waitForRequest((request) => {
+			const matchUrl = request
+				.url()
+				.match(/.*nextgen\.guardianapps\.co\.uk.*lastUpdate=.*/);
+			if (!matchUrl) {
+				return false;
+			}
+			const searchParams = new URLSearchParams(
+				request.url().split('?')[1],
+			);
+			const matchLastUpdate =
+				searchParams.get('lastUpdate') ===
+				'block-62148e2d8f081f9e465d1bb7';
+			return matchLastUpdate;
+		});
+
+		await loadPage(
+			page,
+			`/Article/${blogUrl}?live=true&page=with:block-6214732b8f08f86d89ef68d6&filterKeyEvents=false#liveblog-navigation`,
+		);
+
+		await lastUpdateRequestPromise;
+	});
+
+	test('should handle when the toast is clicked from the second page', async ({
+		page,
+	}) => {
+		await loadPage(
+			page,
+			`/Article/${blogUrl}?live=true&page=with:block-6214732b8f08f86d89ef68d6&filterKeyEvents=false#liveblog-navigation`,
+		);
+		await waitForIsland(page, 'Liveness', { waitFor: 'attached' });
+
+		await page.evaluate(() =>
+			window.scrollTo(0, document.documentElement.scrollHeight),
+		);
+
+		await expectToNotExist(page, `[data-testid="toast"]`);
+
+		await page.evaluate(() => {
+			window.mockLiveUpdate({
+				numNewBlocks: 1,
+				html: '<p>New block 1</p>',
+				mostRecentBlockId: 'abc',
+			});
+		});
+
+		await expectToExist(page, `[data-testid="toast"]`);
+
+		const updateLocator = page
+			.getByText('1 new update')
+			.locator('visible=true');
+		await expect(updateLocator).toBeVisible();
+
+		await updateLocator.click();
+
+		expect(
+			await page.evaluate(() => window.document.location.hash),
+		).toEqual('#maincontent');
```

Comment:
Is this the standard way of assessing the current URL?

https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-url

Reply:
It isn't a standard method but I think it has some advantages as it is a more explicit assertion with exact matching on the location components i.e. `path`, `hash` and `search` rather than a pattern match on the entire url.